src/checksum.c

```diff
@@ -52,28 +52,27 @@ cr_checksum_type(const char *name)
     for (size_t x = 0; x <= len; x++)
         name_lower[x] = tolower(name[x]);
 
-    if (!strncmp(name_lower, "md", 2)) {
-        // MD* family
-//        if (name_lower[2] == '2')
-//            return CR_CHECKSUM_MD2;
-//        else if (name_lower[2] == '5')
-        if (name_lower[2] == '5')
-            return CR_CHECKSUM_MD5;
-    } else if (!strncmp(name_lower, "sha", 3)) {
+    if (!strncmp(name_lower, "sha", 3)) {
         // SHA* family
         char *sha_type = name_lower + 3;
-        if (!strcmp(sha_type, ""))
-            return CR_CHECKSUM_SHA;
-        else if (!strcmp(sha_type, "1"))
-            return CR_CHECKSUM_SHA1;
-        else if (!strcmp(sha_type, "224"))
+        if (!strcmp(sha_type, "224"))
             return CR_CHECKSUM_SHA224;
         else if (!strcmp(sha_type, "256"))
             return CR_CHECKSUM_SHA256;
         else if (!strcmp(sha_type, "384"))
             return CR_CHECKSUM_SHA384;
         else if (!strcmp(sha_type, "512"))
             return CR_CHECKSUM_SHA512;
+#ifdef ENABLE_LEGACY_HASH
+        else if (!strcmp(sha_type, ""))
+            return CR_CHECKSUM_SHA;
+        else if (!strcmp(sha_type, "1"))
+            return CR_CHECKSUM_SHA1;
+    } else if (!strncmp(name_lower, "md", 2)) {
+        // MD* family
+        if (name_lower[2] == '5')
```

Comment:
Consider reordering hashes - the strongest first.
in the future you'll just move the #ifdef up to move weakest among the legacy hashes.